src/praisonai/praisonai/_dev/parity/generator.py

```
@@ -166,179 +255,229 @@ def _get_current_version(self) -> str:
+            return 0
+        print(f"✗ {path} is out of date")
+        return 1
+
+    def _check_markdown(self, path: Path, content: str) -> int:
+        """Compare generated markdown against disk, ignoring the date line."""
+        if not path.exists():
+            print(f"✗ {path} does not exist")
+            return 1
+        with open(path, 'r', encoding='utf-8') as f:
+            existing = f.read()
+        existing_lines = [l for l in existing.split('\n') if not _is_date_line(l)]
```

Ruff (0.16.3)

[error] Ambiguous variable name: `l`

(E741)